quic/s2n-quic-platform/src/io/tokio.rs

```diff
@@ -163,19 +163,20 @@ impl Io {
         #[cfg(s2n_quic_platform_mtu_disc)]
         {
             use std::os::unix::io::AsRawFd;
-            if tx_addr.is_ipv4() {
-                // IP_PMTUDISC_PROBE setting will set the DF (Don't Fragment) flag
-                // while also ignoring the Path MTU. This means packets will not
-                // be fragmented, and the EMSGSIZE error will not be returned for
-                // packets larger than the Path MTU according to the kernel.
-                libc!(setsockopt(
-                    tx_socket.as_raw_fd(),
-                    libc::IPPROTO_IP,
-                    libc::IP_MTU_DISCOVER,
-                    &libc::IP_PMTUDISC_PROBE as *const _ as _,
-                    core::mem::size_of_val(&libc::IP_PMTUDISC_PROBE) as _,
-                ))?;
-            } else {
+
+            // IP_PMTUDISC_PROBE setting will set the DF (Don't Fragment) flag
+            // while also ignoring the Path MTU. This means packets will not
+            // be fragmented, and the EMSGSIZE error will not be returned for
+            // packets larger than the Path MTU according to the kernel.
+            libc!(setsockopt(
+                tx_socket.as_raw_fd(),
+                libc::IPPROTO_IP,
+                libc::IP_MTU_DISCOVER,
+                &libc::IP_PMTUDISC_PROBE as *const _ as _,
+                core::mem::size_of_val(&libc::IP_PMTUDISC_PROBE) as _,
+            ))?;
+
+            if tx_addr.is_ipv6() {
```

Comment:
This should actually be `IPV6_PMTUDISC_PROBE`. See
[IPV6_PMTUDISC_PROBE](https://elixir.bootlin.com/linux/latest/C/ident/IPV6_PMTUDISC_PROBE)

But since the values are equivalent, it shouldn't cause an issue at the moment. 

We might also want to double-check if `size_of_val` is doing the right thing here. This is a `#define` in C, so the value will just be an arbitrary integer type in the generated bindings. But maybe not the same that the API actually expects.

Reply:
According to docs at https://man7.org/linux/man-pages/man7/ip.7.html and demo at https://elixir.bootlin.com/linux/latest/source/tools/testing/selftests/net/udpgso.c#L319 it seems like `IP_PMTUDISC_DO` would also be an option. But as far as I understand for UDP it's equivalent to `IP_PMTUDISC_PROBE `

Reply:
> We might also want to double-check if size_of_val is doing the right thing here.

Ok, I did this one, and it seems good. The values are of type `c_int`: https://docs.rs/libc/latest/libc/constant.IP_PMTUDISC_PROBE.html

And the API expects a pointer to an int (see clarification on https://linux.die.net/man/2/setsockopt)

Reply:
> it seems like IP_PMTUDISC_DO would also be an option

If I'm understanding the docs correctly, this would result in the kernel rejecting MTU probes: 
> The kernel will reject (with EMSGSIZE) datagrams that are bigger than the known path MTU.

Reply:
I think that is only for connected sockets - and even here it seems to be handled euivalently for the 2 options. For both of them the maximum possible IP MTU size seems to be used:

https://elixir.bootlin.com/linux/v5.18.3/source/include/net/ip.h#L436
https://elixir.bootlin.com/linux/v5.18.3/source/net/ipv4/ip_output.c#L1001
https://elixir.bootlin.com/linux/v5.18.3/source/net/ipv4/ip_output.c#L1367